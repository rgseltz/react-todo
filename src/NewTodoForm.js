import React, {useState} from "react";

const NewTodoForm = ({add}) => {
    const INITIAL_STATE = {todo:''}
    const [formData, setFormData] = useState(INITIAL_STATE)
    const handleChange = (evt) => {
        evt.preventDefault()
        const {name, value} = evt.target;
        setFormData({...formData, [name]: value})
    }
    const handleSubmit = evt => {
        evt.preventDefault()
        const {todo} = formData
        add({...formData});
        setFormData('')
    }
    return (
        <div>
            <h2>New Todo</h2>
            <form onSubmit={handleSubmit}>
                    <label htmlFor="todo">Name</label>
                    <input 
                        type="text"
                        name="todo"
                        value = {formData.todo}
                        placeholder="Enter todo here"
                        onChange={handleChange}
                    />
                    <button>Add Todo</button>
            </form>
        </div>
    )
}

export default NewTodoForm;