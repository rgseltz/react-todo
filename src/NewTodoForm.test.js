import React from "react";
import {render, fireEvent} from '@testing-library/react'
import NewTodoForm from "./NewTodoForm";

it('does not crash', () => {
    render(<NewTodoForm />)
})

it('matches snapshot', () => {
    const {asFragment} = render(<NewTodoForm/>);
    expect(asFragment()).toMatchSnapshot();
})