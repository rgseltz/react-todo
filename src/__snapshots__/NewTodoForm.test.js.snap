// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      New Todo
    </h2>
    <form>
      <label>
        Name
      </label>
      <input />
      <button>
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
